library/src/main/java/SpecTreesUnionizer.java

```diff
@@ -187,13 +187,14 @@ private static void processSubtreesWithSameKey(LinkedHashMap<String, Object> map
             valueOfMapToMerge,
             value1);
       } else if (TypeChecker.isObjectList(value1)
```

Comment:
This method is really getting crowded and deeply nested.  Consider breaking apart for clarity.

Reply:
addressed